pkg/sql/sem/builtins/builtins.go

```diff
@@ -8852,6 +8822,89 @@ func strftimeImpl() builtinDefinition {
 	)
 }
 
+func mergeStatsMetadataHelper(
```

Comment:
How do you feel about moving these helper functions to the `sqlstats` package or somewhere similar?

IIUC, we try to keep helper methods outside of `builtins.go`, and use it solely for builtin definitions/overloads.